There's exception if the total sample number cannot be divided by the batch_size.

If the total sample number cannot be divided by the batch_size. we would see bellowing exception.

Caused by: java.lang.IllegalArgumentException: requirement failed: invalid size: 1 == 2
	at scala.Predef$.require(Predef.scala:233)
	at com.intel.analytics.bigdl.tensor.DenseTensor.valueAt(DenseTensor.scala:520)
	at com.intel.analytics.bigdl.nn.ClassNLLCriterion$$anonfun$updateOutput$4.apply(ClassNLLCriterion.scala:100)
	at com.intel.analytics.bigdl.nn.ClassNLLCriterion$$anonfun$updateOutput$4.apply(ClassNLLCriterion.scala:99)
	at com.intel.analytics.bigdl.utils.ThreadPool$$anonfun$invoke$2.apply(ThreadPool.scala:162)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at com.intel.analytics.bigdl.utils.ThreadPool$$anon$1.execute(ThreadPool.scala:203)
	at scala.concurrent.impl.Future$.apply(Future.scala:31)
	at scala.concurrent.Future$.apply(Future.scala:485)
	at com.intel.analytics.bigdl.utils.ThreadPool.invoke(ThreadPool.scala:168)
	at com.intel.analytics.bigdl.nn.ClassNLLCriterion.updateOutput(ClassNLLCriterion.scala:99)
	at com.intel.analytics.bigdl.nn.ClassNLLCriterion.updateOutput(ClassNLLCriterion.scala:60)
	at com.intel.analytics.bigdl.nn.abstractnn.AbstractCriterion.forward(AbstractCriterion.scala:70)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$6$$anonfun$7$$anonfun$apply$2.apply$mcI$sp(DistriOptimizer.scala:205)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$6$$anonfun$7$$anonfun$apply$2.apply(DistriOptimizer.scala:197)
	at com.intel.analytics.bigdl.optim.DistriOptimizer$$anonfun$6$$anonfun$7$$anonfun$apply$2.apply(DistriOptimizer.scala:197)
	at com.intel.analytics.bigdl.utils.ThreadPool$$anonfun$1$$anon$4.call(ThreadPool.scala:112)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)

--- a/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala
+++ b/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala
@@ -83,9 +83,13 @@ class ClassNLLCriterion[@specialized(Float, Double) T: ClassTag]
       output = if (curTarget == -1) ev.zero
       else ev.times(ev.negative(input.valueAt(curTarget)), total_weight)
     } else if (input.dim() == 2) {
+      target.squeeze()
+      require(target.dim() == 1,
+        "ClassNLLCriterion: illegal target! Target should be 1D tensor after squeeze," +
+          s"but target dimension is: ${ target.dim() }, please check your data.")
       val batchSize = input.size(1)
       val targetSize = target.size()
-      target.squeeze()
+
       total_weight = ev.fromType[Int](0)
       output = ev.fromType[Int](0)
 

--- a/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala
+++ b/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala
@@ -88,7 +88,7 @@ class ClassNLLCriterion[@specialized(Float, Double) T: ClassTag]
       target.squeeze()
       require(target.dim() == 1,
         "ClassNLLCriterion: illegal target! Target should be 1D tensor after squeeze," +
-          s"but target dimension is: ${ target.dim() }, please check your data.")
+          s"but target's size is: ${ target.size() }, please check your data.")
 
       total_weight = ev.fromType[Int](0)
       output = ev.fromType[Int](0)

--- a/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala
+++ b/spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala
@@ -83,12 +83,12 @@ class ClassNLLCriterion[@specialized(Float, Double) T: ClassTag]
       output = if (curTarget == -1) ev.zero
       else ev.times(ev.negative(input.valueAt(curTarget)), total_weight)
     } else if (input.dim() == 2) {
+      val batchSize = input.size(1)
+      val targetSize = target.size()
       target.squeeze()
       require(target.dim() == 1,
         "ClassNLLCriterion: illegal target! Target should be 1D tensor after squeeze," +
           s"but target dimension is: ${ target.dim() }, please check your data.")
-      val batchSize = input.size(1)
-      val targetSize = target.size()
 
       total_weight = ev.fromType[Int](0)
       output = ev.fromType[Int](0)

